src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/BasePageObject.java

```diff
@@ -90,23 +90,23 @@ public void waitForWikiaMobileToBeLoaded() {
   }
 
   /**
-   * First waits for spinner to be visible and then waits for spinner to be hidden Spinner presence
+   * First waits for loading overlay (the overlay includes lodaing spinner) to be visible and then waits for spinner to be hidden Spinner presence
    * is optional, when it occurs it must be hidden later
    */
-  public void waitForLoadingSpinnerToFinish() {
+  public void waitForLoadingOverlayToDisappear() {
     boolean spinnerPresent = false;
     try {
       wait.forElementVisible(LOADING_SPINNER_BY, 4, 1000);
       spinnerPresent = true;
     } catch (TimeoutException e) {
-      PageObjectLogging.log("Loading spinner", "is not present", true);
+      PageObjectLogging
+          .log("waitForLoadingOverlayToDisappear", "Lodaing overlay is not visible ", true);
     }
     if (spinnerPresent) {
       wait.forElementNotVisible(LOADING_SPINNER_BY, 4, 3000);
     }
   }
 
-
   //TODO: Remove this and use combination from logUrl
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)